backend/cloudinary/cloudinary.go

```diff
@@ -0,0 +1,626 @@
+	res, err := f.cld.Admin.CreateFolder(ctx, params)
+	f.lastCRUD = time.Now()
+	if err != nil {
+		return err
+	}
+	if res.Error.Message != "" {
+		return errors.New(res.Error.Message)
+	}
+
+	return nil
+}
+
+func (f *Fs) Rmdir(ctx context.Context, dir string) error {
+	params := admin.DeleteFolderParams{Folder: f.ToAssetFolderApi(f.FromStandardFullPath(dir))}
+	res, err := f.cld.Admin.DeleteFolder(ctx, params)
+	f.lastCRUD = time.Now()
+	if err != nil {
+		return err
+	}
+	if res.Error.Message != "" {
+		if strings.HasPrefix(res.Error.Message, "Can't find folder with path") {
+			return fs.ErrorDirNotFound
+		}
+
+		return errors.New(res.Error.Message)
+	}
+
+	return nil
+}
+
+// ------------------------------------------------------------
+
+func (o *Object) Hash(ctx context.Context, ty hash.Type) (string, error) {
+	if ty != hash.MD5 {
+		return "", hash.ErrUnsupported
+	}
+	return o.md5sum, nil
+}
+
+func (o *Object) String() string {
+	if o == nil {
+		return "<nil>"
+	}
+	return o.remote
+}
+
+func (o *Object) Fs() fs.Info {
+	return o.fs
+}
+
+func (o *Object) Remote() string {
+	return o.remote
+}
+
+func (o *Object) ModTime(ctx context.Context) time.Time {
+	return o.modTime
+}
+
+func (o *Object) Size() int64 {
+	return o.size
+}
+
+func (o *Object) Storable() bool {
+	return true
+}
+
+func (o *Object) SetModTime(ctx context.Context, modTime time.Time) error {
+	return fs.ErrorCantSetModTime
+}
+
+func (o *Object) Open(ctx context.Context, options ...fs.OpenOption) (in io.ReadCloser, err error) {
+	var resp *http.Response
+	opts := rest.Opts{
+		Method:  "GET",
+		RootURL: o.url,
+		Options: options,
+	}
+	var offset int64
+	var count int64
+	var key string
+	var value string
+	fs.FixRangeOption(options, o.size)
+	for _, option := range options {
+		switch x := option.(type) {
+		case *fs.RangeOption:
+			offset, count = x.Decode(o.size)
+			if count < 0 {
+				count = o.size - offset
+			}
+			key, value = option.Header()
+		case *fs.SeekOption:
+			offset = x.Offset
+			count = o.size - offset
+			key, value = option.Header()
+		default:
+			if option.Mandatory() {
+				fs.Logf(o, "Unsupported mandatory option: %v", option)
+			}
+		}
+	}
+	if key != "" && value != "" {
+		opts.ExtraHeaders = make(map[string]string)
+		opts.ExtraHeaders[key] = value
+	}
+	// Make sure that the asset is fully available
+	for i := 1; i <= 4; i++ {
+		resp, err = o.fs.srv.Call(ctx, &opts)
+		if err != nil {
+			return nil, fmt.Errorf("failed download of \"%s\": %w", o.url, err)
+		}
+		if count == 0 {
+			break
+		}
+		cl, err := strconv.Atoi(resp.Header.Get("content-length"))
+		if err == nil && count == int64(cl) {
+			break
+		}
+		time.Sleep(time.Duration(i) * time.Second)
+	}
+	return resp.Body, err
+}
+
+func (o *Object) Update(ctx context.Context, in io.Reader, src fs.ObjectInfo, options ...fs.OpenOption) error {
+	srcImmutable := object.NewStaticObjectInfo(o.Remote(), src.ModTime(ctx), src.Size(), true, nil, o.Fs())
```

Comment:
Why are  you making `srcImmutable`? what is wrong with just using `src`?

Reply:
For the wrong reasons... I refactored the needed code, and it looks better. Done